mmdet3d/apis/inference.py

```diff
@@ -128,8 +206,41 @@ def show_result_meshlab(data, result, out_dir):
     if data['img_metas'][0][0]['box_mode_3d'] != Box3DMode.DEPTH:
         points = points[..., [1, 0, 2]]
         points[..., 0] *= -1
-        pred_bboxes = Box3DMode.convert(pred_bboxes,
+        show_bboxes = Box3DMode.convert(pred_bboxes,
                                         data['img_metas'][0][0]['box_mode_3d'],
                                         Box3DMode.DEPTH)
-    show_result(points, None, pred_bboxes, out_dir, file_name, show=False)
+    else:
+        show_bboxes = deepcopy(pred_bboxes)
+    show_result(points, None, show_bboxes, out_dir, file_name, show=False)
+
+    # project 3D bbox to 2D image plane
+    if 'lidar2img' in data['img_metas'][0][0] and \
+            data['img_metas'][0][0]['box_mode_3d'] == Box3DMode.LIDAR:
+        show_bboxes = LiDARInstance3DBoxes(pred_bboxes, origin=(0.5, 0.5, 0))
+        img = mmcv.imread(data['img_metas'][0][0]['filename'])
+
+        show_multi_modality_result(
+            img,
+            None,
+            show_bboxes,
+            data['img_metas'][0][0]['lidar2img'],
+            out_dir,
+            file_name,
+            show=False)
+    elif 'calib' in data.keys() and \
+            data['img_metas'][0][0]['box_mode_3d'] == Box3DMode.DEPTH:
+        show_bboxes = DepthInstance3DBoxes(pred_bboxes, origin=(0.5, 0.5, 0))
+        img = mmcv.imread(data['img_metas'][0][0]['filename'])
+
+        show_multi_modality_result(
+            img,
+            None,
+            show_bboxes,
+            data['calib'][0],
+            out_dir,
+            file_name,
+            depth_bbox=True,
+            img_metas=data['img_metas'][0][0],
+            show=False)
```

Comment:
Should there be an 'else' to avoid/prompt the incorrect input?

Reply:
Add early return for single modality visualization. Raise exceptions if img is provided while projection matrix is not.